src/vhip/mlmodel/gene_features.py

```diff
@@ -160,13 +160,58 @@ def __init__(self, gene_file: str) -> None:
             raise Exception(
                 "Genes file is not provided or empty. Please provide a valid gene file."
             )
-
+        self.id = gene_file
+        self.genes: List[Gene] = []
+        self.skipped_genes: List[str] = []
         readout = read_annotated_genes(gene_file)
-        self.genes: List[Gene] = [
-            Gene(
-                gene_seq=str(readout[0][out]),
-                gene_id=str(readout[1][out]),
-                gene_product=str(readout[2][out]),
+
+        for out in range(len(readout[0])):
+            try:
+                self.genes.append(
+                    Gene(
+                        gene_seq=str(readout[0][out]),
+                        gene_id=str(readout[1][out]),
+                        gene_product=str(readout[2][out]),
+                    )
+                )
+            except Exception:
+                self.skipped_genes.append(str(readout[1][out]))
+        print(f"{len(self.skipped_genes)} genes skipped")
+
```

Comment:
Can you  add in the print statement why those genes were skipped?

Reply:
my intention was because they are not a multiple of codon length. (See Excpetion raised in Gene class.) But what if a different exception occurs? Should I leave as is, to prevent stalling the GeneSet init process if one gene is weird, or should I make my own custom exception class to specifically handle when genes are not multiples of 3?

Reply:
I am not sure what you mean?

My concern right now is that if they were genes skipped, I'd like to know why and which one specifically so I could look into it. So having a method to store those skipped genes and make it accessible would be ideal!

Reply:
Got it. Because readout[1][out] is the gene ids, the gene ids are already stored in the skipped_genes attribute, which can be accessed from any GeneSet object...? In terms of why they were skipped, most of them should be skipped because they are not divisible by codon length. But there is a possibility that some exit the process due to a different exception. I can try to figure out how to log those exceptions and store them.

Reply:
ohhh duh! reading code is harder than writing it haha

What other exceptions do you think you can have?

Reply:
No worries!! Struggling to think of very many. Maybe if the gene sequence is a bunch of numbers instead of a string, there would be no default so Gene init would fail.

Reply:
Actually, I string transform those inputs anyway. So I guess if a sequence is a bunch of numbers, it would get transformed into a Gene. (But it would have so many imprecise codons and get skipped later.) So actually I can't think of any right now

Reply:
So if I can't think of other exceptions, do I need to worry about tracking why genes are skipped? We can assume all are skipped on the basis of codon length divisibility? I could add to the print statement something about the expected average percent to be skipped due to non-divisibility by codon_length? (average 1.12% for viruses, 2.34% for hosts). Would that be good?

Reply:
I like that idea!